editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-helpers.ts

```diff
@@ -238,3 +239,41 @@ export function getInsertionPathForReparentTarget(
   }
   return conditionalClauseInsertionPath(newParent, clause, 'replace')
 }
+
+function areElementsInstancesOfTheSameComponent(
+  firstInstance: ElementInstanceMetadata | null,
+  secondInstance: ElementInstanceMetadata | null,
+): boolean {
+  if (
+    firstInstance == null ||
+    secondInstance == null ||
+    isLeft(firstInstance.element) ||
+    isLeft(secondInstance.element) ||
+    !isJSXElement(firstInstance.element.value) ||
+    !isJSXElement(secondInstance.element.value)
+  ) {
+    return false
+  }
+
+  return jsxElementNameEquals(firstInstance.element.value.name, secondInstance.element.value.name)
+}
+
+export function isElementRenderedBySameComponent(
+  metadata: ElementInstanceMetadataMap,
+  targetPath: ElementPath,
+  instance: ElementInstanceMetadata | null,
+): boolean {
+  if (EP.isEmptyPath(targetPath)) {
+    return false
+  }
+
+  const currentInstance = MetadataUtils.findElementByElementPath(
+    metadata,
+    EP.renderedByParentPath(targetPath),
+  )
+
+  return (
+    areElementsInstancesOfTheSameComponent(currentInstance, instance) ||
+    isElementRenderedBySameComponent(metadata, EP.renderedByParentPath(targetPath), instance)
```

Comment:
I think you can use `EP.getContainingComponent` for the closest component path

Reply:
added in https://github.com/concrete-utopia/utopia/pull/3829/commits/9786303b57ec5eeb33629cf81980a3621313c023